### PR TITLE
ENH: `styler.render.repr` option

### LINKED ISSUE #39911: BUG: DataFrame does not display as LaTeX in Jupyter when it should
- [ X] I have checked that this issue has not already been reported.

- [ X] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

If you run  the following in a Jupyter cell:

```python
import pandas as pd

pd.options.display.latex.repr = True

d = {'one' : [1., 2., 3., 4.],
     'two' : [4., 3., 2., 1.]}
df = pd.DataFrame(d)
df
```
The result is the HTML repr, as usual.  But if you select a column, the Series shows the LaTeX repr, as it should.

```
df['one']
```

But `DataFrame` has `_repr_latex_`, because this works:

```
df._repr_latex_()
```

So it seems like DataFrame is ignoring the `display.latex.repr` option.

#### Problem description

When `display.latex.repr` is True, DataFrames should get represented in LaTeX, not HTML.

#### Expected Output

```
'\\begin{tabular}{lrr}\n\\toprule\n{} &  one &  two \\\\\n\\midrule\n0 &  1.0 &  4.0 \\\\\n1 &  2.0 &  3.0 \\\\\n2 &  3.0 &  2.0 \\\\\n3 &  4.0 &  1.0 \\\\\n\\bottomrule\n\\end{tabular}\n'
```

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : 7d32926db8f7541c356066dcadabf854487738de
python           : 3.7.9.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-72-generic
Version          : #81~16.04.1-Ubuntu SMP Tue Nov 26 16:34:21 UTC 2019
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.2.2
numpy            : 1.19.2
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.0.1
setuptools       : 52.0.0.post20210125
Cython           : None
pytest           : 6.2.2
hypothesis       : None
sphinx           : 3.4.3
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.2
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.20.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.4
numexpr          : 2.7.2
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.6.0
sqlalchemy       : 1.3.23
tables           : 3.6.1
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None
</details>

### PR DESCRIPTION
**Pull request by attack68**
offers option switching between "html" and "latex" output.

adds items to #41395.

also xref #39911 (technically after DataFrame.to_html is deprecated)

- [x] adds option
- [x] adds tests
- [x] updates options.rst docs
- [x] no public method docstrings  
- [x] whatsnew 

update to styler user guide as follow-on after all options are merged.

Screen Shot 2021-08-29 at 10 04 24

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -214,6 +214,8 @@ def _repr_html_(self) -> str:
         """
```
Comment (jreback) [inline pandas/io/formats/style.py:214]: IIRC aren't we supposed to have a ```_repr_latex_``` ?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style.py:214]: I don't know. See if you prefer the new pattern, simple change.

File: pandas/io/formats/style.py [hunk 2]
```diff
@@ -233,10 +233,16 @@ def __init__(
 
     def _repr_html_(self) -> str:
         """
-        Hooks into Jupyter notebook rich display system.
+        Hooks into Jupyter notebook rich display system, which calls _repr_html_ by
+        default if an object is returned at the end of a cell.
         """
+        if get_option("styler.render.repr") == "latex":
```
Comment (jreback) [inline pandas/io/formats/style.py:239]: do we actually still need this? or is this something controlled by the notebook itself?

Comment (attack68) [reply pandas/io/formats/style.py:239]: I tried to look this up, and I may be wrong but I think Notebook uses `_latex_repr_` when converting the notebook to other formats, so it is useful.

When `_repr_html_` and `_repr_latex` are present the notebook still renders the html version. And I can't find docs on native iPython how to configure it to use the other. So the `pandas` options is userfriendly override (per issue)

**Comment by attack68** [comment]
Have refactored this. This is the best, most generic pattern I think. When a _repr_ function returns `None` iPython moves on to the next mimetype. So here we are manually specifying only one _repr_mimetype_ to return a string.

**Comment by jreback** [comment]
thanks I left the display issue open but xref

File: pandas/io/formats/style.py [hunk 3]
```diff
@@ -233,9 +233,17 @@ def __init__(
...
         """
-        return self.to_html()
+        if get_option("styler.render.repr") == "html":
+            return self.to_html()
+        return None
+
+    def _repr_latex_(self) -> str:
+        if get_option("styler.render.repr") == "latex":
+            return self.to_latex()
+        return None
```
Comment (jbrockmendel) [inline pandas/io/formats/style.py:246]: @attack68 i think this is causing the mypy failures on CI

Comment (attack68) [reply pandas/io/formats/style.py:246]: will take a look

**Pull request merged**